pkg/endpoint/bpf.go

```diff
@@ -567,19 +579,25 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string, regenContext *Regene
 		// state with the current program (if it exists) for this endpoint.
 		// GH-3897 would fix this by creating a new map to do an atomic swap
 		// with the old one.
+		stats.mapSync.Start()
 		err := e.syncPolicyMap()
 		if err != nil {
 			e.Unlock()
 			return 0, compilationExecuted, fmt.Errorf("unable to regenerate policy because PolicyMap synchronization failed: %s", err)
 		}
+		stats.mapSync.End()
 
 		// Configure the new network policy with the proxies.
+		stats.proxyPolicyCalculation.Start()
 		if err = e.updateNetworkPolicy(owner, proxyWaitGroup); err != nil {
 			e.Unlock()
 			return 0, compilationExecuted, err
 		}
+		stats.proxyPolicyCalculation.End()
```

Comment:
Should this be put on line 593 if `updateNetworkPolicy` fails?

Reply:
I thought about but it also gives a wrong impression on the duration as it was aborted. I also thought about closing all span brackets in `Total()`.

Reply:
Makes sense; I wasn't too partial one way or the other.